### PR TITLE
Added editing functionality

### PR DESCRIPTION
**Pull request by MariaGuk**

### CODE REVIEW

File: src/api/addNewEmployee.js [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
-  const { data } = await axios.post('https://api-for-masha.herokuapp.com/api/employees', employee );
-  
+  const { data } = await axios.post('https://api-for-masha.herokuapp.com/api/employees', employee);
+
   return data;
 };
 
 const useAddNewEmployee = () => {
+  client.invalidateQueries('employees');
   return useMutation(addNewEmployee)
```
Comment (zhuk-vezunchikz) [inline src/api/addNewEmployee.js]: It makes no sense to invalidate queries before performing the mutation.

https://react-query.tanstack.com/guides/invalidations-from-mutations

File: src/api/deleteEmployee.js [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
   await axios.delete(`https://api-for-masha.herokuapp.com/api/employees/${employeeId}`);
-  
+
   return true;
 };
 
 const useDeleteEmployee = () => {
-  return useMutation(deleteEmployee)
+  client.invalidateQueries('employees');
+  return useMutation(deleteEmployee);
```
Comment (zhuk-vezunchikz) [inline src/api/deleteEmployee.js]: The same thing

File: src/api/editEmployee.js [hunk 3]
```diff
@@ -1,15 +1,16 @@
 import axios from 'axios';
 import { useMutation } from "react-query";
+import { client } from "App"
 
-const editEmployee = async (employeeId) => {
-  const response = await axios.put(`https://api-for-masha.herokuapp.com/api/employees/${employeeId.activeEmployeeId}`, 
-  )
+const editEmployee = async ({ ...data }) => {
```
Comment (zhuk-vezunchikz) [inline src/api/editEmployee.js]: data?

**Review by zhuk-vezunchikz — CHANGES_REQUESTED** [review]

File: src/api/addNewEmployee.js [hunk 4]
```diff
@@ -1,14 +1,19 @@
...
+  const { data } = await axios.post('https://api-for-masha.herokuapp.com/api/employees', employee);
+
   return data;
 };
 
 const useAddNewEmployee = () => {
-  return useMutation(addNewEmployee)
+  return useMutation(addNewEmployee, {
+    onSuccess: () => {
+      client.invalidateQueries('employees');
```
Comment (zhuk-vezunchikz) [inline src/api/addNewEmployee.js:14]: move 'employees' to constants

File: src/api/addNewEmployee.js [hunk 5]
```diff
@@ -1,14 +1,19 @@
 import axios from 'axios';
 import { useMutation } from "react-query";
+import { client } from 'App';
 
 const addNewEmployee = async ({ ...employee }) => {
-  const { data } = await axios.post('https://api-for-masha.herokuapp.com/api/employees', employee );
-  
+  const { data } = await axios.post('https://api-for-masha.herokuapp.com/api/employees', employee);
```
Comment (zhuk-vezunchikz) [inline src/api/addNewEmployee.js:6]: Move url to constants.

File: src/api/deleteEmployee.js [hunk 6]
```diff
@@ -1,14 +1,20 @@
 import axios from 'axios';
 import { useMutation } from "react-query";
+import { client } from 'App';
```
Comment (zhuk-vezunchikz) [inline src/api/deleteEmployee.js:3]: Need an empty line between internal and library imports

File: src/api/deleteEmployee.js [hunk 7]
```diff
@@ -1,14 +1,20 @@
...
+
 
 const deleteEmployee = async (employeeId) => {
   await axios.delete(`https://api-for-masha.herokuapp.com/api/employees/${employeeId}`);
```
Comment (zhuk-vezunchikz) [inline src/api/deleteEmployee.js:7]: url

File: src/api/deleteEmployee.js [hunk 8]
```diff
@@ -1,14 +1,20 @@
...
-  
+
   return true;
 };
 
 const useDeleteEmployee = () => {
-  return useMutation(deleteEmployee)
+  return useMutation(deleteEmployee, {
+    onSuccess: () => {
+      client.invalidateQueries('employees');
```
Comment (zhuk-vezunchikz) [inline src/api/deleteEmployee.js:15]: move

File: src/api/editEmployee.js [hunk 9]
```diff
@@ -1,15 +1,19 @@
 import axios from 'axios';
 import { useMutation } from "react-query";
+import { client } from "App"
 
-const editEmployee = async (employeeId) => {
-  const response = await axios.put(`https://api-for-masha.herokuapp.com/api/employees/${employeeId.activeEmployeeId}`, 
-  )
+const editEmployee = async ({ ...employeeFormData }) => {
+  const { data } = await axios.put(`https://api-for-masha.herokuapp.com/api/employees/${employeeFormData.activeEmployeeId}`, employeeFormData);
```
Comment (zhuk-vezunchikz) [inline src/api/editEmployee.js:6]: move

File: src/api/editEmployee.js [hunk 10]
```diff
@@ -1,15 +1,19 @@
...
 
-  return response.data;
+  return data;
 };
 
 const useEditEmployee = () => {
-  return useMutation(editEmployee)
+  return useMutation(editEmployee, {
+    onSuccess: () => {
+      client.invalidateQueries('employees');
```
Comment (zhuk-vezunchikz) [inline src/api/editEmployee.js:14]: !

File: src/api/getEmployees.js [hunk 11]
```diff
@@ -1,6 +1,5 @@
 import axios from 'axios';
 import { useQuery } from "react-query";
-import { client } from 'App';
 
 const getEmployees = async () => {
   const { data } = await axios.get('https://api-for-masha.herokuapp.com/api/employees');
```
Comment (zhuk-vezunchikz) [inline src/api/getEmployees.js:5]: !

File: src/api/getEmployees.js [hunk 12]
```diff
@@ -12,24 +11,4 @@ const useGetEmployees = () => {
   return useQuery("employees", getEmployees)
```
Comment (zhuk-vezunchikz) [inline src/api/getEmployees.js:11]: !

File: src/components/Card/EmployeeCardContainer.jsx [hunk 13]
```diff
@@ -0,0 +1,65 @@
+import React, { useEffect, useState } from 'react';
+import { useFormik } from 'formik';
+
+import { formValidation } from '../../validation/validation';
+import { useEditEmployee } from '../../api/editEmployee';
+import { useDeleteEmployee } from '../../api/deleteEmployee'
```
Comment (zhuk-vezunchikz) [inline src/components/Card/EmployeeCardContainer.jsx:6]: absolute paths

File: src/pages/Employees/EmployeesContainer.jsx [hunk 14]
```diff
@@ -1,87 +1,57 @@
...
 import LinearProgress from "@material-ui/core/LinearProgress";
 
-import { useGetEmployees, useGetEmployee } from "api/getEmployees";
+import { useGetEmployees } from "api/getEmployees";
 import { useAddNewEmployee } from 'api/addNewEmployee';
-import { useDeleteEmployee } from 'api/deleteEmployee';
-import { useEditEmployee } from 'api/editEmployee';
-import { client } from 'App';
-
+import { formValidation } from '../../validation/validation';
```
Comment (zhuk-vezunchikz) [inline src/pages/Employees/EmployeesContainer.jsx:7]: absolute

File: src/pages/Employees/EmployeesContainer.jsx [hunk 15]
```diff
@@ -1,87 +1,57 @@
...
-      isEmployeeEditing={isEmployeeEditing}
-      isEmployeeDeleting={isEmployeeDeleting}
-      onAddFormSubmit={onAddFormSubmit}
-      onEditFormSubmit={onEditFormSubmit}
-      handleDeleteEmployee={handleDeleteEmployee}
-      currentEmployee={currentEmployee}
-      activeEmployeeId={activeEmployeeId}
-    />
+    <>
+      {employeesProfiles ?
```
Comment (zhuk-vezunchikz) [inline src/pages/Employees/EmployeesContainer.jsx:41]: Use isEmployeeLoading instead of employeesProfiles

File: src/pages/Profile/index.js [hunk 16]
```diff
@@ -1 +1 @@
-export { default } from './ProfileContainer'
\ No newline at end of file
+export { default } from './ProfileContainer.jsx'
```
Comment (zhuk-vezunchikz) [inline src/pages/Profile/index.js:1]: Why did you add this?

File: src/components/Card/EmployeeCardContainer.jsx [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+};
+
+const EmployeeCardContainer = ({ firstName, lastName, email, age, id }) => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const { mutate: editEmployeeMutation } = useEditEmployee();
+
+  const { mutateAsync: deleteEmployeeMutation } = useDeleteEmployee();
+
+  const handleDeleteEmployee = async () => {
```
Comment (zhuk-vezunchikz) [inline src/components/Card/EmployeeCardContainer.jsx:24]: Don't use async/await inside the container

File: src/components/Card/EmployeeCardContainer.jsx [hunk 18]
```diff
@@ -0,0 +1,65 @@
...
+  email: '',
+  age: 0,
+};
+
+const EmployeeCardContainer = ({ firstName, lastName, email, age, id }) => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const { mutate: editEmployeeMutation } = useEditEmployee();
+
+  const { mutateAsync: deleteEmployeeMutation } = useDeleteEmployee();
```
Comment (zhuk-vezunchikz) [inline src/components/Card/EmployeeCardContainer.jsx:22]: Use mutate instead of mutateAsync

File: src/components/Card/EmployeeCardContainer.jsx [hunk 19]
```diff
@@ -0,0 +1,65 @@
...
+    },
+    validationSchema: formValidation
+  });
+
+  useEffect(() => {
+    formik.setValues({ firstName: firstName, lastName: lastName, email: email, age: age });
+  }, [firstName, lastName, email, age]);
+
+  const handleCancel = () => {
+    formik.setValues({ firstName: firstName, lastName: lastName, email: email, age: age });
```
Comment (zhuk-vezunchikz) [inline src/components/Card/EmployeeCardContainer.jsx:43]: Why we set values when closing modal?

Comment (MariaGuk) [reply src/components/Card/EmployeeCardContainer.jsx:43]: Because if we close the modal window, but we don't want to save the changes, but just want to close modal, then when the modal window is reopened, data, entered earlier, will be displayed. 
Perhaps there is a better way to avoid such an error.
